drivers/i2c/i2c_omap.c

```diff
@@ -0,0 +1,947 @@
+/* Copyright (C) 2024 BeagleBoard.org Foundation
+ * Copyright (C) 2024 Dhruv Menon <dhruvmenon1104@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_omap_i2c
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+#include <zephyr/drivers/i2c.h>
+#include "i2c_bitbang.h"
+
+LOG_MODULE_REGISTER(omap_i2c, CONFIG_I2C_LOG_LEVEL);
+
+#define I2C_OMAP_TIMEOUT    100U
+/* OCP_SYSSTATUS bit definitions */
+#define SYSS_RESETDONE_MASK BIT(0)
+
+#define I2C_BITRATE_FAST     400000
+#define I2C_BITRATE_STANDARD 100000
+
+/* I2C Registers */
+#define I2C_REVNB_LO        0x0  /* Revision number low */
+#define I2C_REVNB_HI        0x4  /* Revision number high */
+#define I2C_SYSC            0x10 /* System configuration */
+#define I2C_EOI             0x20 /* End of interrupt */
+#define I2C_IRQSTATUS_RAW   0x24 /* Per-event raw interrupt status vector */
+#define I2C_IRQSTATUS       0x28 /* Per-event enabled interrupt status vector */
+#define I2C_IRQENABLE_SET   0x2c /* Per-event interrupt enable bit set */
+#define I2C_IRQENABLE_CLR   0x30 /* Per-event interrupt clear bit clear */
+#define I2C_WE              0x34 /* Wakeup enable */
+#define I2C_DMARXENABLE_SET 0x38 /* Per-event DMA RX enable set */
+#define I2C_DMATXENABLE_SET 0x3c /* Per-event DMA TX enable set */
+#define I2C_DMARXENABLE_CLR 0x40 /* Per-event DMA RX enable clear */
+#define I2C_DMATXENABLE_CLR 0x44 /* Per-event DMA TX enable clear */
+#define I2C_DMARXWAKE_EN    0x48 /* DMA RX wake enable */
+#define I2C_DMATXWAKE_EN    0x4c /* DMA TX wake enable */
```

Comment:
Lot of unused defines? is this copied from some existing TI driver? then you need to retain the original copyright as well.